src/app/modules/item/components/item-children-edit/item-children-edit.component.ts

```diff
@@ -1,22 +1,51 @@
 import { Component, Input, OnChanges } from '@angular/core';
 import { ItemChild } from './item-children';
+import { ItemData } from '../../services/item-datasource.service';
+import { GetItemChildrenService } from '../../http-services/get-item-children.service';
+import { Subscription } from 'rxjs';
 
 @Component({
   selector: 'alg-item-children-edit',
   templateUrl: './item-children-edit.component.html',
   styleUrls: [ './item-children-edit.component.scss' ]
 })
 export class ItemChildrenEditComponent implements OnChanges {
+  @Input() itemData?: ItemData;
 
-  @Input() data: ItemChild[] = [
-    { id: '100', title: 'First item', order: 1 },
-    { id: '50', title: 'Second item', order: 2 },
-  ];
+  state: 'loading' | 'error' | 'empty' | 'ready' = 'ready';
+  data: ItemChild[] = [];
 
-  constructor() {}
+  private subscription?: Subscription;
+
+  constructor(
+    private getItemChildrenService: GetItemChildrenService,
+  ) {}
 
   ngOnChanges(): void {
-    this.data = this.data.sort((a, b) => a.order - b.order);
+    this.reloadData();
+  }
+
+  reloadData(): void {
+    if (this.itemData?.currentResult) {
+      this.state = 'loading';
+      this.subscription?.unsubscribe();
+      this.subscription = this.getItemChildrenService
+        .get(this.itemData.item.id, this.itemData.currentResult.attemptId)
+        .subscribe(children => {
+          this.data = children.map((child, idx) => ({
+            id: child.id,
+            title: child.string.title,
+            order: idx
+          }));
```

Comment:
I prefer you do this "map" in a Rx pipe map function. (I know we do not do it everywhere in the existing code)
This way, if there is a problem (for instance there is no "string" in "child", it ends up as a proper error (in the block below) instead of crash the whole thing.

Reply:
Done